### ISSUE TITLE
Get-UnityLicense fails because the versioning is malformed

### ISSUE THREAD

**Issue by StephenHodgson** [microsoft/unitysetup.powershell#142]
Unity's license file doesn't always have well formed version definitions.

My personal license says it's for `2017.2.0` which is invalid to the `UnityLicense` this module expects.

![image](https://user-images.githubusercontent.com/13334553/50939944-cacf9480-1433-11e9-90bf-b80931725308.png)

```
Cannot convert value "2017.2.0" to type "UnityVersion". Error: "Invalid unity version: 2017.2.0"
At C:\Users\shodgson\Documents\WindowsPowerShell\Modules\UnitySetup\4.0.97\UnitySetup.psm1:1016 char:24
+ ...  'Serial' = ConvertTo-SecureString $licenseSerial -AsPlainText -Force ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvalidCastConstructorException
```

**Comment by StephenHodgson** [comment]
Digging in a bit deeper I noticed our [regex expression](https://github.com/Microsoft/unitysetup.powershell/blob/develop/UnitySetup/UnitySetup.psm1#L150) when parsing the version out may need to be updated a bit.

https://regex101.com/r/7AGvpa/2

Adding an `|` can match either the short version for long version (Also, seems to be faster too!)

Likely we'll also need to update the match count check

**Comment by chrisdjali-wrld3d** [comment]
This is still an issue and should be reopened. I just got an error saying `Cannot convert value "2017.2.0" to type "UnityVersion". Error: "Invalid unity version: 2017.2.0"` when running `Get-UnityLicense`.

**Comment by deitry** [comment]
This is also still an issue for both my Pro and Personal licenses

**Comment by deitry** [comment]
It seems 2017.2.0 is being applied when license is activated via Unity Hub. If license is activated via command line with specific Unity.exe, license has that version

**Comment by StephenHodgson** [comment]
Because it's never gonna be addressed, and I've moved onto other tools

**Comment by jwittner** [comment]
I reopened, as this project is still actively maintained even if Stephen is moving on. If someone want to make a PR to support short versions in the licensing I'm happy to review. =)

**Comment by jwittner** [comment]
Fix available in 5.5.156-develop.
